src/Corax/IndexSearcher/IndexSearcher.Search.cs

```diff
@@ -81,30 +81,39 @@ public IQueryMatch SearchQuery(string field, string searchTerm, Constants.Search
             if (mode.HasFlag(Constants.Search.SearchMatchOptions.StartsWith | Constants.Search.SearchMatchOptions.EndsWith))
                 mode = Constants.Search.SearchMatchOptions.Contains;
 
-            var termForTree = encoded.Slice(tokens[0].Offset, (int)tokens[0].Length);
+            int termStart = tokens[0].Offset;
+            int termLength = (int)tokens[0].Length;
             if (manuallyCutWildcards)
             {
                 if (mode != Constants.Search.SearchMatchOptions.TermMatch)
                 {
-                    if (mode is Constants.Search.SearchMatchOptions.Contains && termForTree.Length <= 2)
+                    if (mode is Constants.Search.SearchMatchOptions.Contains && termLength <= 2)
                         throw new InvalidDataException(
                             "You are looking for an empty term. Search doesn't support it. If you want to find empty strings, you have to write an equal inside WHERE clause.");
-                    else if (termForTree.Length == 1)
+                    if (termLength == 1)
                         throw new InvalidDataException("You are looking for all matches. To retrieve all matches you have to write `true` in WHERE clause.");
                 }
 
-                termForTree = mode switch
+                (int startIncrement, int lengthIncrement) = mode switch
                 {
-                    Constants.Search.SearchMatchOptions.StartsWith => termForTree.Slice(1),
-                    Constants.Search.SearchMatchOptions.EndsWith => termForTree.Slice(0, termForTree.Length - 1),
-                    Constants.Search.SearchMatchOptions.Contains => termForTree.Slice(1, termForTree.Length - 2),
-                    Constants.Search.SearchMatchOptions.TermMatch => termForTree,
+                    Constants.Search.SearchMatchOptions.StartsWith => (1,-1),
+                    Constants.Search.SearchMatchOptions.EndsWith => (0,-1),
+                    Constants.Search.SearchMatchOptions.Contains => (1,-3),
+                    Constants.Search.SearchMatchOptions.TermMatch => (0,0),
```

Comment:
What is the meaning of those?

Reply:
To be true, I have no idea. I just got rid of the Span slicing. @maciejaszyk can you document that?

Reply:
Well, because the analyzer doesn't return us TermVectors I've to find it on my own. I use <Lowercase, Whitespace> analyzer to split the queried sentence and check if the user used the `*` prefix (or/and) suffix. This gives us information what match to use for specific term (`TermMatch` / `StartsWithMatch` / `EndsWithMatch` or `ContainsMatch`).

Reply:
What are those numbers, such as `(1,3)` ?

Reply:
Can we use some const with meaningful names here?

Reply:
@maciejaszyk

Reply:
@redknightlois the second number is `lengthIncrement`. But I don't see any usage of that. Is that correct?

Reply:
Great catch... And that is a bug!!! Now, how it didnt trigger any failure is beyond interesting. We have a clear hole in the testing suit in there.